### PR TITLE
feat: add C implementation for `stats/base/dists/uniform/entropy`

### LINKED ISSUE #3826: [RFC]: Add C implementation for `@stdlib/stats/base/dists/uniform/entropy`
### Description

This RFC proposes adding a C implementation, including a Node.js native addon, C benchmarks, and C examples, for the [`@stdlib/stats/base/dists/uniform/entropy`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/@stdlib/stats/base/dists/uniform/entropy) package.

When adding support, the following tasks should be completed:

1. A C source implementation should be added to a `src` folder within the package's directory.
2. A native add-on interface should be added in order to call the C implementation from JavaScript.
3. A JavaScript file should be added to the `lib` folder to allow benchmarking and unit testing the native implementation from JavaScript.
4. A new test file should be created which includes unit tests testing against expected behavior.
5. C benchmarks should be added which measure performance for the C source implementation.
6. A file containing C examples should be added to demonstrate example usage.
7. JavaScript benchmarks should be added which measure performance for the native add-on.
8. Build and configuration files should be added to allow compiling and running the C source implementation using project `make` commands (as documented below).
9. The README should be updated to include documentation for the added C API.

To provide a concrete example of what a PR adding the desired method should contain, see https://github.com/stdlib-js/stdlib/pull/3354, which is a PR adding a C implementation for the CDF of an arcsine distribution. This should provide an idea of what is expected.

#### Prerequisites

1. Review JavaScript implementation in `lib/main.js` and its required modules to identify all `stdlib` functions used.
2. Verify C implementations exist for each required function.

### Related Issues

#3355

### Questions

No.

### Other

Once the implementation is ready including C examples and benchmarks, we should be able to run the following `make` commands.

#### Build native add-on

```bash
NODE_ADDONS_PATTERN="@stdlib/stats/base/dists/uniform/entropy" make install-node-addons
```

#### Run C examples

```bash
make examples-c-files FILES="$(pwd)/lib/node_modules/@stdlib/stats/base/dists/uniform/entropy/examples/c/example.c"
```

#### Run C benchmarks

```bash
make benchmark-c-files FILES="$(pwd)/lib/node_modules/@stdlib/stats/base/dists/uniform/entropy/benchmark/c/benchmark.c"
```

#### Run JavaScript benchmarks

```bash
make benchmark-javascript-files FILES="$(pwd)/lib/node_modules/@stdlib/stats/base/dists/uniform/entropy/benchmark/benchmark.native.js"
```

#### Run all tests

```bash
make TESTS_FILTER=".*/@stdlib/stats/base/dists/uniform/entropy/.*" test
```

### Notes

- If you are interested in contributing a PR which addresses this RFC and are a **first-time contributor** or not yet familiar with our project conventions, please minimize the submission of LLM-generated code. Please consult our [contributing guidelines](https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md) and the associated [development guide](https://github.com/stdlib-js/stdlib/blob/develop/docs/development.md). Failure to respect project conventions will result in your PR being rejected without review. Thank you for understanding!

### Checklist

- [x] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [x] Searched for existing issues and pull requests.
- [x] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by sethdivyansh**
Resolves #3826

## Description

- adds C implementation for @stdlib/stats/base/dists/uniform/entropy along with relevant tests, benchmarks and examples

This pull request:

-  progresses: #3826 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

File: lib/node_modules/@stdlib/stats/base/dists/uniform/entropy/README.md [hunk 1]
```diff
@@ -133,6 +133,101 @@ for ( i = 0; i < 10; i++ ) {
...
+
+<section class="usage">
+
+### Usage
+
+```c
+#include "stdlib/stats/base/dists/uniform/entropy.h"
+```
+
+#### stdlib_base_dists_uniform_entropy( n, p )
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/uniform/entropy/README.md]:
```suggestion
#### stdlib_base_dists_uniform_entropy( a, b )
```

File: lib/node_modules/@stdlib/stats/base/dists/uniform/entropy/README.md [hunk 2]
```diff
@@ -133,6 +133,101 @@ for ( i = 0; i < 10; i++ ) {
...
+
+Evaluates the [entropy][entropy] of a [Uniform][uniform-distribution] distribution with `a` the minimum support and `b` the maximum support.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/uniform/entropy/README.md]:
```suggestion
Evaluates the [entropy][entropy] of a [uniform][uniform-distribution] distribution with `a` the minimum support and `b` the maximum support.
```

File: lib/node_modules/@stdlib/stats/base/dists/uniform/entropy/src/main.c [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include "stdlib/stats/base/dists/uniform/entropy.h"
+#include "stdlib/math/base/assert/is_nan.h"
+#include "stdlib/math/base/special/ln.h"
+#include <math.h>
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/uniform/entropy/src/main.c]: Don't think `math.h` is needed here:

```suggestion
```

File: lib/node_modules/@stdlib/stats/base/dists/uniform/entropy/src/main.c [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+/**
+* Returns the entropy of a uniform distribution.
+*
+* @param n   minimum support
+* @param p   maximum support
+* @returns	 entropy
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/uniform/entropy/src/main.c:29]:
```suggestion
* @return   entropy
```

File: lib/node_modules/@stdlib/stats/base/dists/uniform/entropy/src/main.c [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+* var v = entropy( 4.0, 12.0 );
+* // returns ~2.079
+*
+* @example
+* var v = entropy( 1.0, -0.1 );
+* // returns NaN
+*
+* @example
+* var v = entropy( NaN, 2.0 );
+* // returns NaN
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/uniform/entropy/src/main.c]: One example is enough for the C files:

```suggestion
* @param n   minimum support
* @param p   maximum support
* @return    entropy
*
* @example
* var v = entropy( 0.0, 1.0 );
* // returns 0.0
```

File: lib/node_modules/@stdlib/stats/base/dists/uniform/entropy/src/main.c [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+*/
+double stdlib_base_dists_uniform_entropy( const double a, const double b ) {
+
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/uniform/entropy/src/main.c]:
```suggestion
```

File: lib/node_modules/@stdlib/stats/base/dists/uniform/entropy/examples/c/example.c [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+* limitations under the License.
+*/
+
+#include "stdlib/stats/base/dists/uniform/entropy.h"
+#include <stdlib.h>
+#include <stdio.h>
+
+static double random_uniform( const double min, const double max ) {
+    double v = (double)rand() / ( (double)RAND_MAX + 1.0 );
+    return min + ( v*(max-min) );
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/uniform/entropy/examples/c/example.c]:
```suggestion
	double v = (double)rand() / ( (double)RAND_MAX + 1.0 );
	return min + ( v*(max-min) );
```

**Review by Planeshifter — CHANGES_REQUESTED** [review]
Thanks, @sethdivyansh!

Left comments that will have to be addressed before merging, but otherwise looks fine to me.

**Comment by sethdivyansh** [comment]
Done with the changes. Please review.

File: lib/node_modules/@stdlib/stats/base/dists/uniform/entropy/src/main.c [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+* // returns ~2.079
+*
+* @example
+* var v = entropy( 1.0, -0.1 );
+* // returns NaN
+*
+* @example
+* var v = entropy( NaN, 2.0 );
+* // returns NaN
+*/
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/uniform/entropy/src/main.c]:
```suggestion
* @example
* double v = entropy( 0.0, 1.0 );
* // returns 0.0
*/
```

**Review by Planeshifter — APPROVED** [review]
Thanks for the edits!

Made one more change but let's get your PR in now. 🚀

**Pull request merged**